braindecode/preprocessing/windowers.py

```diff
@@ -628,11 +628,26 @@ def _create_windows_from_events(
             metadata=metadata,
             description=ds.description,
         )
-    # add window_kwargs and raw_preproc_kwargs to windows dataset
+    
+    # Serialize the windowing operation
+    windowing_kwargs_dict = window_kwargs[0][1]
+    windowing_serialized = _serialize_windowing_operation(
+        create_windows_from_events.__name__, 
+        windowing_kwargs_dict
+    )
+    
+    # Propagate raw_preproc_kwargs and append windowing operation
+    if hasattr(ds, "raw_preproc_kwargs"):
+        # Copy existing preprocessing steps and add windowing operation
+        windows_ds.raw_preproc_kwargs = list(ds.raw_preproc_kwargs)
+        windows_ds.raw_preproc_kwargs.append(windowing_serialized)
+    else:
+        # Just add windowing operation
+        windows_ds.raw_preproc_kwargs = [windowing_serialized]
+    
+    # Keep window_kwargs for backwards compatibility
     setattr(windows_ds, "window_kwargs", window_kwargs)
-    kwargs_name = "raw_preproc_kwargs"
-    if hasattr(ds, kwargs_name):
-        setattr(windows_ds, kwargs_name, getattr(ds, kwargs_name))
```

Comment:
Oh I see that the feature I initially thought was missing is actually already implemented 

Maybe the implementation is nicer now...